include/nbl/builtin/hlsl/shapes/triangle.hlsl

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2018-2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_SHAPES_TRIANGLE_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SHAPES_TRIANGLE_INCLUDED_
+
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace shapes
+{
+
+namespace util
+{
+  template <typename float_t>
+  vector<float_t, 3> GetAngleWeight(const vector<float_t, 3>& e1, const vector<float_t, 3>& e2, const vector<float_t, 3>& e3)
```

Comment:
cool idea to kick it to HLSL, but to be HLSL, you need to use `NBL_CONST_REF_ARG` because `const T&` won't compile as HLSL